src/main/java/org/jitsi/videobridge/LipSyncHack.java

```diff
@@ -540,28 +618,166 @@ public void close()
     /**
      *
      */
-    private static class Injection
+    private class Injection
+        implements RecurringRunnable
     {
         /**
          * Ctor.
          *
-         * @param maxSeqNum
-         * @param maxTs
+         * @param receiveVideoSSRC
+         */
+        Injection(long receiveVideoSSRC)
+        {
+            this.receiveVideoSSRC = receiveVideoSSRC;
+        }
+
+        /**
+         *
+         */
+        final long receiveVideoSSRC;
+
+        /**
+         * The maximum sequence number (mod 2^16) that has been sent out by
+         * this injector instance.
+         */
+        int maxSeqNum = -1;
+
+        /**
+         * The maximum RTP timstamp (mod 2^32) that has been sent out by
+         * this injector instance.
+         */
+        long maxTs = -1;
+
+        /**
+         * A boolean indicating whether or not this instance is still injecting
+         * packets.
+         */
+        boolean running = true;
+
+        /**
+         * The last time in milliseconds at which {@link #run} was invoked.
+         */
+        long lastRunMs = -1;
+
+        /**
+         *
          */
-        public Injection(int maxSeqNum, long maxTs)
+        @Override
+        public long getTimeUntilNextRun()
         {
-            this.maxSeqNum = maxSeqNum;
-            this.maxTs = maxTs;
+            if (!running)
+            {
+                return Long.MAX_VALUE;
+            }
+
+            long timeSinceLastProcess
+                = Math.max(System.currentTimeMillis() - lastRunMs, 0);
+
+            return Math.max(PERIOD_MS - timeSinceLastProcess, 0);
         }
 
         /**
          *
          */
-        final int maxSeqNum;
+        @Override
+        public void run()
+        {
+            if (!running)
+            {
+                return;
+            }
+
+            lastRunMs = System.currentTimeMillis();
+
+            MediaStream stream = dest.getStream();
+            if (stream == null)
+            {
+                return;
+            }
+
+            synchronized (this)
```

Comment:
gotcha.  i think it'd be good to put this guarantee in the java docs for stop so it's more clear (i.e.: "after stop is called, guaranteed that the injection logic is stopped and maxSeqNum and maxTs will not be further modified" or something)